internal/http/services/owncloud/ocdav/tus.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2018-2020 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package ocdav
+
+import (
+	"net/http"
+	"path"
+
+	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	typespb "github.com/cs3org/go-cs3apis/cs3/types/v1beta1"
+	"github.com/cs3org/reva/pkg/appctx"
+	tusd "github.com/tus/tusd/pkg/handler"
+)
+
+func (s *svc) handleTusPost(w http.ResponseWriter, r *http.Request, ns string) {
+	ctx := r.Context()
+	log := appctx.GetLogger(ctx)
+
+	w.Header().Add("Access-Control-Allow-Headers", "Tus-Resumable, Upload-Length, Upload-Metadata, If-Match")
+	w.Header().Add("Access-Control-Expose-Headers", "Tus-Resumable, Location")
+
+	// Test if the version sent by the client is supported
+	// GET methods are not checked since a browser may visit this URL and does
+	// not include this header. This request is not part of the specification.
+	if r.Header.Get("Tus-Resumable") != "1.0.0" {
+		w.WriteHeader(http.StatusPreconditionFailed)
+		return
+	}
+	if r.Header.Get("Upload-Length") == "" {
+		w.WriteHeader(http.StatusPreconditionFailed)
+		return
+	}
+
+	// read filename from metadata
+	meta := tusd.ParseMetadataHeader(r.Header.Get("Upload-Metadata"))
+	if meta["filename"] == "" {
+		w.WriteHeader(http.StatusPreconditionFailed)
+		return
+	}
+
+	// append filename to current dir
+	fn := path.Join(ns, r.URL.Path, meta["filename"])
```

Comment:
@PVince81 this is where the wrong path is build when the ns is set to /home but the url path is taken from the dav/files endpoint, because it starts with the username ... but /home paths don't need the username. 💥 

If you could make Phoenix send tho POST to the old webdav endpoint this would work for now ...